ArduCopter/mode.h

```diff
@@ -17,7 +17,11 @@ class Mode {
     virtual bool init(bool ignore_checks) {
         return true;
     }
-    virtual void run() = 0;
+    void run_mode() {
+        surface_tracking.invalidate_for_logging();  // invalidate surface tracking alt, flight mode will set to true if used
```

Comment:
I think adding another "run" is just confusing for developers.. "what's the difference between run_mode()" and run?" they will ask.  We could instead perhaps make use of an exit function or something to invalidate it or perhaps use a timer so if it's not updated for 20ms the logging stops using it.

Reply:
I've renamed `run_mode` to `update`.

It's unclear to me where any "exit" function would be called - we don't want to have every mode have to remember to make that call.